quantus/helpers/model/model_interface.py

```diff
@@ -105,3 +105,36 @@ def get_random_layer_generator(self):
         set it to 'independent'. For bottom-up order, set it to 'bottom_up'.
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def get_hidden_representations(
+        self,
+        x: np.ndarray,
+        layer_names: Optional[List[str]] = None,
+        layer_indices: Optional[List[int]] = None,
+    ) -> np.ndarray:
+        """
+        Computes models internal representation of input x.
```

Comment:
Rewrite to:

Compute the model's internal representation of input x.
In practice, this means, executing a forward pass and then, capturing the output of layers (of interest).
As the exact definition of  "internal model representation" is left out in the original paper (see:  https://arxiv.org/pdf/2203.06877.pdf), we make the implementation flexible.
It is up to the user whether all layers are used, or specific ones should be selected.
The user can therefore select a layer by providing 'layer_names' (exclusive) or 'layer_indices'.

Reply:
done